go/cmd/vtexplain/vtexplain.go

```diff
@@ -29,62 +28,46 @@ import (
-	sqlFileFlag        = flag.String("sql-file", "", "Identifies the file that contains the SQL commands to analyze")
-	schemaFlag         = flag.String("schema", "", "The SQL table schema")
-	schemaFileFlag     = flag.String("schema-file", "", "Identifies the file that contains the SQL table schema")
-	vschemaFlag        = flag.String("vschema", "", "Identifies the VTGate routing schema")
-	vschemaFileFlag    = flag.String("vschema-file", "", "Identifies the VTGate routing schema file")
-	ksShardMapFlag     = flag.String("ks-shard-map", "", "JSON map of keyspace name -> shard name -> ShardReference object. The inner map is the same as the output of FindAllShardsInKeyspace")
-	ksShardMapFileFlag = flag.String("ks-shard-map-file", "", "File containing json blob of keyspace name -> shard name -> ShardReference object")
-	numShards          = flag.Int("shards", 2, "Number of shards per keyspace. Passing --ks-shard-map/--ks-shard-map-file causes this flag to be ignored.")
-	executionMode      = flag.String("execution-mode", "multi", "The execution mode to simulate -- must be set to multi, legacy-autocommit, or twopc")
-	replicationMode    = flag.String("replication-mode", "ROW", "The replication mode to simulate -- must be set to either ROW or STATEMENT")
-	normalize          = flag.Bool("normalize", false, "Whether to enable vtgate normalization")
-	outputMode         = flag.String("output-mode", "text", "Output in human-friendly text or json")
-	dbName             = flag.String("dbname", "", "Optional database target to override normal routing")
-	plannerVersionStr  = flag.String("planner-version", "", "Sets the query planner version to use when generating the explain output. Valid values are V3 and Gen4")
-
-	// vtexplainFlags lists all the flags that should show in usage
-	vtexplainFlags = []string{
-		"output-mode",
-		"planner-version",
-		"normalize",
-		"shards",
-		"replication-mode",
-		"schema",
-		"schema-file",
-		"sql",
-		"sql-file",
-		"vschema",
-		"vschema-file",
-		"ks-shard-map",
-		"ks-shard-map-file",
-		"dbname",
-		"queryserver-config-passthrough-dmls",
-	}
+	sqlFlag            string
+	sqlFileFlag        string
+	schemaFlag         string
+	schemaFileFlag     string
+	vschemaFlag        string
+	vschemaFileFlag    string
+	ksShardMapFlag     string
+	ksShardMapFileFlag string
+	numShards          int
+	executionMode      string
+	replicationMode    string
+	normalize          bool
+	outputMode         string
+	dbName             string
+	plannerVersionStr  string
 )
 
 func init() {
-	logger := logutil.NewConsoleLogger()
-	flag.CommandLine.SetOutput(logutil.NewLoggerWriter(logger))
-	_flag.SetUsage(flag.CommandLine, _flag.UsageOptions{
-		FlagFilter: func(f *flag.Flag) bool {
-			for _, name := range vtexplainFlags {
-				if f.Name == name {
-					return true
-				}
-			}
-
-			return false
-		},
+	servenv.OnParse(func(fs *pflag.FlagSet) {
```

Comment:
We could have used `OrParseFor("vtexplain", func...)` but I don't think it matters too much since it is the cmd file

Reply:
@ajm188 wdyt?

Reply:
i have been pushing toward using simply `OnParse` when you're in a cmd; the argument i have used is since it's a `main` package, no one can import you, so there's no material difference between a flag registered to "a binary" (since there is just the one), and "all binaries".